File: pubnub-java/src/main/java/com/pubnub/internal/endpoints/presence/WhereNowImpl.java
package com.pubnub.internal.endpoints.presence;

import com.pubnub.internal.InternalPubNubClient;
import com.pubnub.internal.endpoints.DelegatingEndpoint;
import com.pubnub.internal.models.consumer.presence.PNWhereNowResult;
import lombok.Setter;
import lombok.experimental.Accessors;

@Setter
@Accessors(chain = true, fluent = true)
public class WhereNowImpl extends DelegatingEndpoint<PNWhereNowResult> implements com.pubnub.api.endpoints.presence.WhereNow {

    private String uuid;

    public WhereNowImpl(InternalPubNubClient pubnub) {
        super(pubnub);
        uuid = pubnub.getConfiguration().getUuid();
    }

    @Override
    protected WhereNow createAction() {
        return pubnub.whereNow(uuid);
    }
}

File: pubnub-kotlin/src/main/kotlin/com/pubnub/internal/endpoints/presence/WhereNowImpl.kt
package com.pubnub.internal.endpoints.presence

import com.pubnub.api.DelegatingEndpoint
import com.pubnub.api.endpoints.remoteaction.ExtendedRemoteAction
import com.pubnub.api.endpoints.remoteaction.MappingRemoteAction
import com.pubnub.internal.PubNubImpl
import com.pubnub.internal.models.consumer.presence.PNWhereNowResult
import com.pubnub.internal.models.toApi

/**
 * @see [PubNubImpl.whereNow]
 */
class WhereNowImpl internal constructor(whereNow: WhereNowEndpoint) :
    DelegatingEndpoint<com.pubnub.api.models.consumer.presence.PNWhereNowResult, PNWhereNowResult>(whereNow),
    WhereNowInterface by whereNow,
    com.pubnub.api.endpoints.presence.WhereNow {
        override fun convertAction(
            remoteAction: ExtendedRemoteAction<PNWhereNowResult>,
        ): ExtendedRemoteAction<com.pubnub.api.models.consumer.presence.PNWhereNowResult> {
            return MappingRemoteAction(
                remoteAction,
                PNWhereNowResult::toApi,
            )
        }
    }
